### PR TITLE
Infinite scroll of map and panning out of map boundaries . #2248

### LINKED ISSUE #2248: Infinite scroll of map and panning out of map boundaries .
# Describe the problem (or idea) :  

The maps (for example at http://publiclab.org/people) on scrolling left/right adds new layer and the **map scrolling becomes infinite** . 

And also we can **drag the map out of its boundaries** . 

#### Lets make boundaries for the map so that we can restrict out of bound scrolling and dragging of map . 

giphy

### PR DESCRIPTION
**Pull request by sagarpreet-chadha**
Fixes #2248 😄 🎈 .
@jywarren kindly see this . **This file has become very large , should i add more comments here ?** What do you think ?

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

Thanks!

### CODE REVIEW

**Comment by sagarpreet-chadha** [comment]
The gif is here : [https://github.com/publiclab/plots2/issues/2248#issuecomment-366022363](https://github.com/publiclab/plots2/issues/2248#issuecomment-366022363) . Thanks !

**Comment by jywarren** [comment]
Very nice. Yes, perhaps we should break this out into a file in `/app/assets/javascripts/` too? See how you can parameterize anything we need to pass in from Ruby into a function constructor:

https://github.com/publiclab/plots2/blob/e15927ab3af9d674c1b94ff76c814e8c1548d7f8/app/assets/javascripts/tagging.js#L1

It could be `setupMapLayers()` or `setupLeafletMap()` maybe, or maybe one for the map, one for the layers? 

This is very cool! Great work!

**Comment by sagarpreet-chadha** [comment]
Thank you @jywarren  !! On it !!

File: app/assets/javascripts/leaflet_helper.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    var baseMaps = {
+      "Default": normal_layer,
+    };
+    var overlayMaps = {
+      "Skynet": layerGroup    // we can add more layers here !
+    }; 
+    L.control.layers(baseMaps , overlayMaps).addTo(map);
+  }
+
+  function setupFullScreen(map , lat , lon) {
```
Comment (Gauravano) [inline app/assets/javascripts/leaflet_helper.js]: Really liked the way you removed the script from view and added and divided in functions in separate file :100:

Comment (sagarpreet-chadha) [reply app/assets/javascripts/leaflet_helper.js]: Thank you @Gauravano 😄 !

**Comment by jywarren** [comment]
Fantastic! Great work here. 👍 🎆 🚀

**Pull request merged**